crates/rspack_plugin_javascript/src/visitors/tree_shaking.rs

```diff
@@ -16,26 +19,33 @@ pub fn tree_shaking_visitor<'a>(
   module_graph: &'a ModuleGraph,
   module_id: Identifier,
   used_symbol_set: &'a HashSet<SymbolRef>,
-  top_level_mark: Mark,
   side_effects_free_modules: &'a IdentifierSet,
   module_item_map: &'a IdentifierMap<Vec<ModuleItem>>,
-  helper_mark: Mark,
+  extra_mark: ExtraMark,
+  include_module_ids: &'a IdentifierSet,
+  options: Arc<CompilerOptions>,
 ) -> impl Fold + 'a {
   TreeShaker {
     module_graph,
     decl_mappings,
     module_identifier: module_id,
     used_symbol_set,
-    top_level_mark,
     module_item_index: 0,
     insert_item_tuple_list: Vec::new(),
     side_effects_free_modules,
     last_module_item_index: 0,
     module_item_map,
-    helper_mark,
+    extra_mark,
+    include_module_ids,
+    options,
   }
 }
 
+pub struct ExtraMark {
```

Comment:
Not sure this is a good name. If these marks are extra， what marks aren't?

Reply:
Because I don't want to  pass the mark as arguments one by one, combining multiple marks as one struct could reduce the count of arguments

Reply:
> Because I don't want to pass the mark as arguments one by one, combining multiple marks as one struct could reduce the count of arguments

Yeah. I agree with this. It's just the name of the struct is kind of ambiguous.

Reply:
yeah, I will pick a better name.